common/conf.hpp

```diff
@@ -92,9 +92,128 @@ class CmdlineOptionsIterator {
+	string ArtScriptsPath = path::Join(DataStore, "scripts");
+	string ModulesWorkPath = path::Join(DataStore, "modules/v3");
+	string BootstrapArtifactFile = path::Join(DataStore, "bootstrap.mender");
+	string FallbackConfFile = path::Join(DataStore, "mender.conf");
+
+	string KeyFile = "mender-agent.pem";
+
+public:
+	string GetPathConfDir() const {
+		return PathConfDir;
+	}
+	void SetPathConfDir(const string &conf_dir) {
+		this->PathConfDir = conf_dir;
+		this->ConfFile = path::Join(PathConfDir, "mender.conf");
+		this->RootfsScriptsPath = path::Join(PathConfDir, "scripts");
+	}
+
+	string GetPathDataDir() const {
+		return PathDataDir;
+	}
+	void SetPathDataDir(const string &PathDataDir) {
+		this->PathDataDir = PathDataDir;
+		this->IdentityScript =
+			path::Join(PathDataDir, "identity", "mender-device-identity");
+		this->ModulesPath = path::Join(PathDataDir, "modules/v3");
+	}
+
+	string GetDataStore() const {
+		return DataStore;
+	}
+	void SetDataStore(const string &DataStore) {
+		this->DataStore = DataStore;
+		this->FallbackConfFile = path::Join(DataStore, "mender.conf");
+		this->ArtScriptsPath = path::Join(DataStore, "scripts");
+		this->ModulesWorkPath = path::Join(DataStore, "modules/v3");
+		this->BootstrapArtifactFile = path::Join(DataStore, "bootstrap.mender");
+		return;
+	}
+
+	string GetKeyFile() const {
+		return KeyFile;
+	}
+	void SetKeyFile(const string &KeyFile) {
+		this->KeyFile = KeyFile;
+	}
+
+
+	string GetConfFile() const {
+		return ConfFile;
+	}
+	void SetConfFile(const string &ConfFile) {
+		this->ConfFile = ConfFile;
+	}
+
+	string GetFallbackConfFile() const {
+		return FallbackConfFile;
+	}
+	void SetFallbackConfFile(const string &FallbackConfFile) {
+		this->FallbackConfFile = FallbackConfFile;
+	}
+
+	string GetIdentityScript() const {
+		return IdentityScript;
+	}
+	void SetIdentityScript(const string &IdentityScript) {
+		this->IdentityScript = IdentityScript;
+	}
+
+	string GetArtScriptsPath() {
+		return ArtScriptsPath;
+	}
+	void SetArtScriptsPath(const string &ArtScriptsPath) {
+		this->ArtScriptsPath = ArtScriptsPath;
+	}
+
+	string GetRootfsScriptsPath() const {
+		return RootfsScriptsPath;
+	}
+	void SetRootfsScriptsPath(const string &RootfsScriptsPath) {
+		this->RootfsScriptsPath = RootfsScriptsPath;
+	}
+
+	string GetModulesPath() const {
+		return ModulesPath;
+	}
+	void SetModulesPath(const string &ModulesPath) {
+		this->ModulesPath = ModulesPath;
+	}
+
+	string GetModulesWorkPath() const {
+		return ModulesWorkPath;
+	}
+	void SetModulesWorkPath(const string &ModulesWorkPath) {
+		this->ModulesWorkPath = ModulesWorkPath;
+	}
+
+	string GetBootstrapArtifactFile() const {
+		return BootstrapArtifactFile;
+	}
+	void SetBootstrapArtifactFile(const string &BootstrapArtifactFile) {
+		this->BootstrapArtifactFile = BootstrapArtifactFile;
+	}
+};
+
 class MenderConfig : public cfg_parser::MenderConfigFromFile {
 public:
-	string data_store_dir = paths::DefaultDataStore;
+	Paths default_config {};
```

Comment:
Why call it `default_config` and not `paths`?

Reply:
moved to `paths`